nemo_skills/evaluation/metrics/base.py

```diff
@@ -29,6 +30,92 @@ def get_metrics(self):
     def reset(self):
         pass
 
+    @abc.abstractmethod
+    def get_prediction_results(self, prediction):
+        pass
+
+    def get_majority_at_k(
+        self, agg_mode_dict, predicted_answers, pred_keys=None, predictions=None, prediction_results=None
```

Comment:
For agg_mode_dict and self._get_best_prediction, I would just use those methods from main class. So you can define get_best_prediction to have the general logic in the base class and then subclasses can override. And agg_mode_dict is anyway used everywhere, so let's just move it to the base class and use directly instead of passing as a parameter?